actionpack/lib/action_controller/metal/strong_parameters.rb

```diff
@@ -1088,23 +1239,45 @@ def specify_numeric_keys?(filter)
         end
       end
 
-      def each_element(object, filter, &block)
+      # When an array is expected, you must specify an array explicitly
+      # using the following format:
+      #
+      #     params.expect(comments: [[:flavor]])
+      #
+      # Which will match only the following array formats:
+      #
+      #     { pies: [{ flavor: "rhubarb" }, { flavor: "apple" }] }
+      #     { pies: { "0" => { flavor: "key lime" }, "1" =>  { flavor: "mince" } } }
+      #
+      # Previously, arrays were specified the same way as hashes:
```

Comment:
Maybe change to "When using `permit`, arrays are specified...". Also use `permit` in the below example, since `expect` never allowed non-explicit arrays in the first place.